src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -15,6 +15,11 @@
 #include <sof/audio/component.h>
 #include <sof/ut.h>
 #include <sof/lib/memory.h>
+#include "module_interface.h"
+
+#if CONFIG_INTEL_MODULES
+#include "iadk_modules.h"
+#endif
```

Comment:
Its better to have the ```include "iadk_modules.h"``` in the Intel module client code as this is still a dependency on generic.h i.e. it should be in the C/C++ Intel code that uses the IADK module API.

e.g. in the Intel client code
```c
#include <module/generic.h>
#include <module/iadk_modules.h>
```

Reply:
There is one problem. Library manager creates new comp_driver object after module load and needs to assign its ops at this point. It will load IADK or SOF loadable modules but at some point it needs to select correct set of ops callbacks. See this: https://github.com/thesofproject/sof/blob/96618de55e0417f976761d69eb020359d7912216/src/library_manager/lib_manager.c#L369

We could add some function call to generic.h, that will do it inside generic code.  But still generic will need to include iadk_modules.h functions. Don't know yet how to separate these two better.

Reply:
I understand your point, but even module_shim_new() will be in generic it has to refer to **struct module_interface iadk_interface()**   which is iadk_modules.c specific. To be honest I don't know yet how to avoid this dependency. For built in modules you can create and initialize component driver at built time. For loadable You will got UUID at loading time so it has to be called at IPC service context.